src/easier68k/core/enum/ea_mode_bin.py

```diff
@@ -7,31 +7,47 @@
 
 class EAModeBinary:
     # Data register direct
-    M_DRD = 0b000
+    MODE_DRD = 0b000
 
     # Address register direct
-    M_ARD = 0b001
+    MODE_ARD = 0b001
 
     # Address register indirect
-    M_ARI = 0b010
+    MODE_ARI = 0b010
 
     # Address register indirect + post increment
-    M_ARIPI = 0b011
+    MODE_ARIPI = 0b011
 
     # Address register indirect + pre decrement
-    M_ARIPD = 0b100
+    MODE_ARIPD = 0b100
 
     # Immediate
-    M_IMM = 0b111
-    XN_IMM = 0b100
+    MODE_IMM = 0b111
+    REGISTER_IMM = 0b100
 
     # Absolute long address
-    M_ALA = 0b111
-    XN_ALA = 0b001
+    MODE_ALA = 0b111
+    REGISTER_ALA = 0b001
 
     # Absolute word address
-    M_AWA = 0b111
-    XN_AWA = 0b000
+    MODE_AWA = 0b111
+    REGISTER_AWA = 0b000
+
+    # currently missing the offset modes
+    VALID_DEST_EA_MODES = [MODE_DRD, MODE_ARI, MODE_ARIPI, MODE_ARIPD, MODE_ALA, MODE_AWA]
+    
+    # currently missing the offset modes
+    VALID_SRC_EA_MODES = VALID_DEST_EA_MODES + [MODE_ARD, MODE_IMM]
+    
+    # what the hell should I name this?
+    # it's the valid destination registers for the 111 mode
+    # currently missing the offset modes
+    VALID_DEST_EA_111_REGISTERS = [REGISTER_ALA, REGISTER_AWA]
+    
+    # what the hell should I name this?
+    # it's the valid source registers for the 111 mode
+    # currently missing the offset modes
+    VALID_SRC_EA_111_REGISTERS = VALID_DEST_EA_111_REGISTERS + [REGISTER_IMM]
 
     @staticmethod
     def parse_from_ea_mode_mfirst(mode: EAMode) -> str:
```

Comment:
and I would move this to the module and outside of the class

Reply:
Done